changelogs/fragments/51149-allow-merging-module-defaults-groups.yaml

```diff
@@ -0,0 +1,9 @@
+minor_changes:
+  - module_defaults - Add a new MODULE_DEFAULTS_MERGE configuration option.
+deprecated_features:
+  - module_defaults - Overwriting module_defaults set in an outer scope by default is deprecated. In 2.12
+    the default behavior will be to merge module_defaults set in an inner scope with those set in the outer
+    scope. Set MODULE_DEFAULTS_MERGE to true to enable merging before 2.12.
+  - module_defaults - Using an empty dictionary to clear previous module defaults for a module/group is
+    deprecated and will be removed in 2.12. Instead, set the special '__clear__' key to true for the
```

Comment:
instead of adding a new token, can't we use overwrite + explicitly null value?

having Sentinel class now lets us use null/None as a user provided 'nullifier'